File: cpp/grains/grains.cpp
#include "grains.h"
#include <cmath>

namespace grains {

long long unsigned int square(int square_number) {
    return pow(2, square_number - 1);
}

long long unsigned int total() {
    int sum{0};
    for (int i = 0; i < 64; i++) {
        sum += square(i);
    }
    return sum;
}

}  // namespace grains

File: java/grains/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger grainsOnSquare(final int square) {
        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }
        return new BigInteger("0").setBit(square - 1);
    }

    BigInteger grainsOnBoard() {
        BigInteger total = new BigInteger("0");
        for (int i = 1; i <= 64; i++) {
            total = total.add(grainsOnSquare(i));
        }
        return total;
    }

}
